src/common/websocket/index.ts

```diff
@@ -34,22 +33,59 @@ export class ApiManager {
         return ApiManager.instance
     }
 
+    public static readonly READY_STATE_KEY = 'websocket_ready_state'
+
+    private setReadyStateInSessionStorage(state: number) {
+        if (isBrowser()) {
+            sessionStorage.setItem(ApiManager.READY_STATE_KEY, state.toString())
+        }
+    }
+
     public init(lang?: string) {
         if (!this.ready) {
             if (!this.socket) {
+                console.log('WS connecting...')
                 const language = lang === 'ach' ? getCrowdin() : lang?.replace('-', '_')
                 const socket_url = getSocketURL()
                 const app_id = getAppId()
                 const websocket_connection_url = `${socket_url}?app_id=${app_id}&l=${language}&brand=${brand_name.toLowerCase()}`
 
                 this.socket = new WebSocket(websocket_connection_url)
+                this.setReadyStateInSessionStorage(this.socket?.readyState)
             }
             this.derivApi = new DerivAPIBasic({ connection: this.socket })
-            this.registerKeepAlive()
+            this.socket.addEventListener('open', () => {
+                console.log('WS connected.')
+                this.setReadyStateInSessionStorage(this?.socket?.readyState)
+            })
+
+            this.socket.addEventListener('close', () => {
+                console.log('WS closed')
+                this.derivApi.disconnect()
+                this.ready = null
+                sessionStorage.removeItem(ApiManager.READY_STATE_KEY)
```

Comment:
You should have kept the implementation of your storage inside the `setReadyStateInSessionStorage`. With your approach, if you change it from session-storage to local-storage you need to update all places. 

So, if you had something like 
```
// PSEUDO

private setReadyStateInSessionStorage(key: string, value: number) {
   if (value == null) sessionStorage.removeItem(key)
   else sessionStorage. setItem(key, value)
}
```

you should both _set_ and _unset_ things via this function.

Reply:
removed session storage so resolving this.